test/test_helper.rb

```diff
@@ -20,6 +20,17 @@
 require 'google-id-token'
 GoogleSignIn::Identity.validator = GoogleIDToken::Validator.new(x509_cert: GOOGLE_X509_CERTIFICATE)
 
+# Suppress incorrect OAuth2 client warning about having both an access token
+# and an ID token. They aren't interchangeable. And ID token is returned with
+# OIDC scoped requests and is used for authentication, whereas the access token
+# is used for authorization.
+module SuppressOAuthExtraTokensWarning
+  def from_hash(client, hash)
+    new client, hash.fetch("access_token"), hash.except("access_token")
+  end
+end
+OAuth2::AccessToken.singleton_class.prepend SuppressOAuthExtraTokensWarning
```

Comment:
OAuth2 gem incorrectly treats access tokens and ID tokens as interchangeable, emitting a warning if both are present. Both are meant to be present in OIDC: the ID token for authentication and the access token to authorize access to whatever other scopes were requested.

This hacks the implementation to bypass the warning.

Reply:
References https://gitlab.com/oauth-xx/oauth2/-/issues/628